x/ugov/client/cli/query.go

```diff
@@ -121,3 +122,33 @@ func QueryInflationCyleEnd() *cobra.Command {
 
 	return cmd
 }
+
+// QueryTokenBalances creates the Query/TokenBalances CLI.
+func QueryTokenBalances() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "token-balances [denom]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Queries for all account addresses that own a particular token denomination.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+			pageReq, err := client.ReadPageRequest(cmd.Flags())
+			if err != nil {
+				return err
+			}
+			queryClient := ugov.NewQueryClient(clientCtx)
+			resp, err := queryClient.TokenBalances(cmd.Context(), &ugov.QueryTokenBalances{
+				Denom:      args[0],
+				Pagination: pageReq,
+			})
+			return cli.PrintOrErr(resp, err, clientCtx)
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+	flags.AddPaginationFlagsToCmd(cmd, "token-balances")
+
+	return cmd
+}
```

Comment:
do we need CLI? I think web-grpc is enough

Reply:
we will keep this